pathway-analysis/02-gene_set_enrichment_analysis.Rmd

```diff
@@ -111,77 +126,121 @@ dge_results_df <- readr::read_tsv(dge_results_file)
+This will keep the first row with the duplicated value thus keeping the row with the highest absolute value of the log2 fold change.
+
+```{r filter_dge}
+filtered_dge_df <- dge_results_df %>%
+  # Sort so that the highest absolute values of the log2 fold change are at the
+  # top
+  dplyr::arrange(dplyr::desc(abs(log2FoldChange))) %>%
+  # Filter out the duplicated rows using `dplyr::distinct()`
+  dplyr::distinct(gene_symbol, .keep_all = TRUE)
+```
+
+Let's see what happened to our duplicate identifiers.
+
+```{r show_filtered_dge, live = TRUE}
+# Subset to & arrange by gene symbols that were duplicated in the original 
+# data frame of results
+filtered_dge_df %>%
+  dplyr::filter(gene_symbol %in% duplicated_gene_symbols) %>%
+  dplyr::arrange(gene_symbol)
+```
+
+Now we're ready to prep our pre-ranked list for GSEA.
+
+### Pre-ranked list
+
+The `GSEA()` function takes a pre-ranked (sorted) named vector of statistics, where the names in the vector are gene identifiers.
 This is step 1 -- gene-level statistics.
 
 ```{r lfc_vector}
-lfc_vector <- dge_results_df$log2FoldChange
-names(lfc_vector) <- dge_results_df$entrez_id
+lfc_vector <- filtered_dge_df %>%
+  dplyr::pull(log2FoldChange, name = gene_symbol)
 lfc_vector <- sort(lfc_vector, decreasing = TRUE)
 ```
 
+Let's look at the top ranked values.
+
 ```{r head_lfc, live = TRUE}
 # Look at first entries of the log2 fold change vector
 head(lfc_vector)
 ```
 
+And the bottom of the list.
+
 ```{r tail_lfc, live = TRUE}
 # Look at the last entries of the log2 fold change vector
 tail(lfc_vector)
 ```
 
-## GSEA
-
-![](diagrams/subramanian_fig1.jpg)
-
-**Figure 1. [Subramanian, Tamayo *et al.* (2005)](https://doi.org/10.1073/pnas.0506580102).**
+## Run GSEA
 
-The enrichment score (ES) for a pathway, a pathway-level statistic, is calculated using our gene-level statistics.
-Genes are ranked from most highly positive to most highly negative and weighting them according to their gene-level statistic.
-A running score is calculated by starting with the most highly ranked genes and increasing the score when a gene is in the pathway and decreasing the score when a gene is not in the pathway.
-The ES is the maximum deviation from zero.
-Significance is assessed by generating a null distribution by sampling random gene sets of the same size and an FDR (false discovery rate) value is calculated to account for multiple hypothesis testing.
-([Subramanian, Tamayo *et al.* 2005](https://doi.org/10.1073/pnas.0506580102); [Korotkevich, Sukhov, and Sergushichev. 2019](https://doi.org/10.1101/060012)).
+Now for the analysis!
 
 We can use the `GSEA()` function to perform GSEA with any generic set of gene sets, but there are several functions for using specific, commonly used gene sets (e.g., `gseKEGG()`).
 
 ```{r run_gsea}
 gsea_results <- GSEA(geneList = lfc_vector,  # ordered ranked gene list
-                     nPerm = 1000,  # number of permutations
```

Comment:
Just noted you took this out... is it the default, or are we just happy with the default?

Reply:
New version of `clusterProfiler` deprecated this argument (I think? at least it yells at you), and the default for `fgseaMultilevel` (what happens under the hood) is still 1000.